dashboard/src/components/service/registry-checking.service.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2015-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+'use strict';
+
+import { GlobalWarningBannerService } from './global-warning-banner.service';
+import { CheWorkspace } from '../api/workspace/che-workspace.factory';
+import { CheBranding } from '../branding/che-branding';
+
+/**
+ * This service does HEAD requests to plugin and devfile registries. In case if at least one of requests fails, a banner with a warning message will be show.
+ * @author Oleksii Kurinnyi
+ */
+export class RegistryCheckingService {
+
+  static $inject = [
+    '$http',
+    '$q',
+    'cheBranding',
+    'cheWorkspace',
+    'globalWarningBannerService',
+  ];
+
+  private $http: ng.IHttpService;
+  private $q: ng.IQService;
+  private cheWorkspace: CheWorkspace;
+  private globalWarningBannerService: GlobalWarningBannerService;
+  private cheBranding: CheBranding;
+
+  private headers: { [name: string]: string; };
+
+  constructor(
+    $http: ng.IHttpService,
+    $q: ng.IQService,
+    cheBranding: CheBranding,
+    cheWorkspace: CheWorkspace,
+    globalWarningBannerService: GlobalWarningBannerService,
+  ) {
+    this.$http = $http;
+    this.$q = $q;
+    this.cheBranding = cheBranding;
+    this.cheWorkspace = cheWorkspace;
+    this.globalWarningBannerService = globalWarningBannerService;
+
+    this.headers = { 'Authorization': undefined };
+
+    const REGISTRY_CERTIFICATE_ERROR = `Unable to load plugins and/or devfiles. Your ${this.cheBranding.getName()} may be using a self-signed certificate. To resolve this issue, try to import the servers CA certificate into your browser, as described in <a href="${this.cheBranding.getDocs().certificate}" target="_blank">docs</a>. After importing the certificate, refresh the page.`;
+
+    this.cheWorkspace.fetchWorkspaceSettings()
+      .then(settings => [
+        `${settings.cheWorkspaceDevfileRegistryUrl}/devfiles/index.json`,
+        `${settings.cheWorkspacePluginRegistryUrl}/plugins/`
+      ])
+      .then(locations => this.$q.all(
+        locations.map(location => this.checkPossibleCertificateIssue(location))
+      ))
+      .catch(() => this.globalWarningBannerService.addMessage(REGISTRY_CERTIFICATE_ERROR));
```

Comment:
Is it shown when devfile/plugin registry returned 501/404.... ?

Reply:
Thanks for pointing this out, I updated corresponding code to show the banner only when request fails because of untrusted host.

Reply:
![Screenshot_20200318_131617](https://user-images.githubusercontent.com/5887312/76955393-c39af500-691a-11ea-842b-e48ab9c0ad9d.png)

I still see the banner in case non-tls but service unavailability issue

Reply:
Unfortunately, these requests end with status `(failed)` instead of one of HTTP statuses, e.g 404 or 501. So we can’t separate requests failed because of untrusted host from ones to unavailable service.

<img width="706" alt="Screenshot 2020-03-19 at 11 44 36" src="https://user-images.githubusercontent.com/16220722/77053600-3ff30e00-69d7-11ea-90f3-66a75669cb23.png">

Reply:
=( Then I'm OK with leaving as is.